core/blockchain.go

```diff
@@ -1971,6 +1986,18 @@ func (bc *BlockChain) reorg(oldBlock, newBlock *types.Block) error {
 	if newBlock == nil {
 		return fmt.Errorf("invalid new chain")
 	}
+
+	if oldBlock.NumberU64() == newBlock.NumberU64() &&
+		oldBlock.Hash() == newBlock.Hash() {
+		log.Error("Reorg called for same blocks",
+			"oldnum", oldBlock.Number(),
+			"oldhash", oldBlock.Hash().Hex(),
+			"newnum", newBlock.Number(),
+			"newhash", newBlock.Hash().Hex(),
+		)
+		return fmt.Errorf("reorg new block same as old")
```

Comment:
Actual change. Have `blockchain.reorg` method return an error if, after some preliminary logic to derive "old" and "new" blocks, it turns out that they are actually identical. This will handle one of the edge cases that the following error return would handle otherwise.